src/app.rs

```diff
@@ -24,6 +24,61 @@ use crate::{
 /// Apps are equipped with a handle to their own state (`Data`), which is available to all endpoints.
 /// This is a "handle" because it must be `Clone`, and endpoints are invoked with a fresh clone.
 /// They also hold a top-level router.
+///
+/// # Examples
+///
+/// You can start a simple Tide application that listens for `GET` requests at path `/hello`
+/// on `127.0.0.1:7878` with:
+///
+/// ```rust, no_run
+/// #![feature(async_await)]
+///
+/// let mut app = tide::App::new(());
+/// app.at("/hello").get(async || "Hello, world!");
+/// app.serve("127.0.0.1:7878")
+/// ```
+///
+/// Create a new `App` with a `Data` handle and make a clone of that handle available to endpoints:
+///
+/// ```rust, no_run
+/// #![feature(async_await, futures_api)]
+///
+/// use std::sync::Arc;
+/// use std::sync::Mutex;
+/// use tide::AppData;
```

Comment:
Is it possible to elaborate on `AppData`? Right now it seems a little opaque as to why the `Database` I passed as a parameter to `tide::App::new` is accessible as an `AppData` to my endpoint.

Reply:
Thanks for the feedback. I've updated the description for the example - let me know what you think.

Reply:
Yes, this explanation is really good!

Reply:
Thanks!